### ISSUE TITLE
False positive error when solving for TypeVar

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#8048]
This issue is related to #6207 but involves a change in the argument order.

```python
from typing import Callable

def func8[T](a: T, b: Callable[[list[T]], None]) -> T:
    return a

def func9(v: Callable[[list[int]], None]):
    func8(1, v)
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.374.
